gef.py

```diff
@@ -1936,13 +1936,10 @@ def gef_pybytes(x: str) -> bytes:
 @lru_cache()
 def which(program: str) -> pathlib.Path:
     """Locate a command on the filesystem."""
-    for path in os.environ["PATH"].split(os.pathsep):
-        dirname = pathlib.Path(path)
-        fpath = dirname / program
-        if os.access(fpath, os.X_OK):
-            return fpath
-
-    raise FileNotFoundError(f"Missing file `{program}`")
+    try:
+        return pathlib.Path(shutil.which(program))
+    except TypeError:
+        raise FileNotFoundError(f"Missing file `{program}`")
```

Comment:
Why catching `TypeError`?
Also, always better to fail first.

```suggestion
    res = shutil.which(program)
    if not res:
        raise FileNotFoundError(f"Missing file `{program}`")
    return pathlib.Path(res)
```

Reply:
Yeah, explicitly checking for None feels cleaner. If @Grazfather agrees, I can revert and apply this suggestion instead.

Reply:
fine by me